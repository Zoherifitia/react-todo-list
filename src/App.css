.App {
  text-align: center;
}
.input_todo{
  height: 30px;
  border-radius: 5px;
}
h1{
  text-align: center;
  color: white;
  background-color: #888844;
}
.container {
  text-align: center;
  display: flex;
  width: 60%;
  height: auto;
  background-color: #888844;
  margin-left: 20%;
}
.todolist{
  border: 2px solid white;
  width: 50%;
  height: auto;
  border-radius: 3px;
}
.donelist{
  border: 2px solid white;
  width: 50%;
  height: auto;
  border-radius: 3px;
}
.input{
  width: 90%;
  padding: 1% 0%;
  margin: 2% 0% 0% 5%;
  border: 1px outset white;
  border-radius: 6px;
  text-align: initial;
}
.donelist div{
  width: 90%;
  padding: 1% 0% 1% 1% ;
  margin: 2% 0% 4% 5%;
  border: 1px inset white;
  border-radius: 6px;
  text-align: initial;
}
.donelist h3{
  margin-bottom: 9%;
}

