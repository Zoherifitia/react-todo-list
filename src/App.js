import './App.css';
import React, { useState } from "react";
import { FormTodo } from './components/FormTodo';
import { Todo } from './components/Todo';

function App() {
  const [todos, setTodos] = React.useState([
    {
      text: "This is a sample todo",
      isDone: false
      }
    ]);
  const [dones,setDones] = useState([]);
  
  const addTodo = text => {
    const newTodos = [...todos, { text }];
    setTodos(newTodos);
  };
  
  const markTodo = index => {
    const newTodos = [...todos];
    const todo = newTodos[index];
    todo.isDone = true;
    setDones([...dones, todo]);
    newTodos.splice(index, 1);
    setTodos(newTodos);
  };

  return (
    <div className="app">
      <h1>Todo List</h1>
      <div className='container'>
        <div className='todolist'>
          <FormTodo addTodo={addTodo} />
          <div >
            {todos.map((todo,index) => (
             <p>
                  <Todo
                  key={Todo+index}
                  index={index}
                  todo={todo}
                  markTodo={markTodo}
                  />
              </p>
            ))}
          </div>
        </div>
        <div className='donelist'>
            <h3>done</h3>
            <p>this todo is done</p>
            <p>
              {dones.map(dones => {
                return(
                  <div>{dones.text}</div>
                )
              })} 
            </p>
        </div>
      </div>
    </div>
  );
}

export default App;