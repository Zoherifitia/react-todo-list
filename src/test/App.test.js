import renderer from 'react-test-renderer';
import React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../App';
import { renderIntoDocument } from 'react-dom/test-utils';

/*test('app header rendering', () => {
  render(<App />);
  const linkElement = screen.getByText(/Todo List/i)
  expect(linkElement).toBeInTheDocument();
});*/

describe('render app', () => {
  it('renders correctly', () => {
    const app = jest.fn();
    const tree = renderer
      .create(<App />)
      .toJSON();
    console.log(tree)
    expect(tree).toMatchSnapshot();
  });
});

 