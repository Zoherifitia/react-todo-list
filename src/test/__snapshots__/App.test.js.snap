// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app renders correctly 1`] = `
<div
  className="app"
>
  <h1>
    Todo List
  </h1>
  <div
    className="container"
  >
    <div
      className="todolist"
    >
      <form
        onSubmit={[Function]}
      >
        <h3>
          Todo
        </h3>
        <input
          className="input_todo"
          onChange={[Function]}
          placeholder="Add new todo"
          type="text"
          value=""
        />
      </form>
      <div>
        <p>
          <div
            className="todo"
          >
            <div
              className="input"
            >
              <input
                id={0}
                onClick={[Function]}
                type="checkbox"
              />
              <label
                htmlFor={0}
              />
              <span>
                This is a sample todo
              </span>
            </div>
          </div>
        </p>
      </div>
    </div>
    <div
      className="donelist"
    >
      <h3>
        done
      </h3>
      <p>
        this todo is done
      </p>
      <p />
    </div>
  </div>
</div>
`;
